packages/core/js-sdk/src/store/index.ts

```diff
@@ -0,0 +1,267 @@
+import { Client } from "../client"
+
+export class Store {
+  private client: Client
+
+  constructor(client: Client) {
+    this.client = client
+  }
+
+  public region = {
+    list: async (
+      queryParams?: Record<string, any>,
+      headers?: Record<string, any>
+    ) => {
+      return this.client
+        .fetch(`/store/regions`, { query: queryParams, headers })
+        .then((resp) => resp.json())
+    },
+    retrieve: async (
+      id: string,
+      queryParams?: Record<string, any>,
+      headers?: Record<string, any>
+    ) => {
+      return this.client
+        .fetch(`/store/regions/${id}`, { query: queryParams, headers })
+        .then((resp) => resp.json())
+    },
+  }
+
+  public collection = {
+    list: async (
+      queryParams?: Record<string, any>,
+      headers?: Record<string, any>
+    ) => {
+      return this.client
+        .fetch(`/store/collections`, { query: queryParams, headers })
+        .then((resp) => resp.json())
+    },
+    retrieve: async (
+      id: string,
+      queryParams?: Record<string, any>,
+      headers?: Record<string, any>
+    ) => {
+      return this.client
+        .fetch(`/store/collections/${id}`, { query: queryParams, headers })
+        .then((resp) => resp.json())
+    },
+  }
+
+  public category = {
+    list: async (
+      queryParams?: Record<string, any>,
+      headers?: Record<string, any>
+    ) => {
+      return this.client
+        .fetch(`/store/product-categories`, { query: queryParams, headers })
+        .then((resp) => resp.json())
+    },
+    retrieve: async (
+      id: string,
+      queryParams?: Record<string, any>,
+      headers?: Record<string, any>
+    ) => {
+      return this.client
+        .fetch(`/store/product-categories/${id}`, {
+          query: queryParams,
+          headers,
+        })
+        .then((resp) => resp.json())
+    },
+  }
+
+  public product = {
```

Comment:
I think these can be extracted into their own files, rather than implementing them inline, but I'll refrain from doing that until we decide on the few points I listed in the description.